### PR TITLE
allow gradient transform to handle asymmetrical shapes

### PR DESCRIPTION
**Pull request by greerdv**
## What does this PR do?
Allows gradient transforms to handle shapes which are not symmetrical about their centre e.g. shapes with translation offsets, polygon prisms. Related to #13398.

## How was this PR tested?
Manual testing in editor and running unit tests.

### CODE REVIEW

File: Gems/GradientSignal/Code/Source/Components/GradientTransformComponent.cpp [hunk 1]
```diff
@@ -83,7 +84,9 @@ namespace GradientSignal
                     ->Attribute(AZ::Edit::Attributes::ReadOnly, &GradientTransformConfig::IsAdvancedModeReadOnly)
                     ->DataElement(0, &GradientTransformConfig::m_bounds, "Bounds", "Local (untransformed) bounds of a box used to remap, clamp, wrap, scale incoming coordinates")
                     ->Attribute(AZ::Edit::Attributes::ReadOnly, &GradientTransformConfig::IsBoundsReadOnly)
-                    
+                    ->DataElement(0, &GradientTransformConfig::m_bounds, "Center", "Local (untransformed) center of a box used to remap, clamp, wrap, scale incoming coordinates")
```
Comment (mbalfour-amzn) [inline Gems/GradientSignal/Code/Source/Components/GradientTransformComponent.cpp:87]: If this stays here, it should use m_center, not m_bounds.

Comment (greerdv) [reply Gems/GradientSignal/Code/Source/Components/GradientTransformComponent.cpp:87]: good catch, thanks

**Review by mbalfour-amzn — APPROVED** [review]

File: Gems/GradientSignal/Code/Include/GradientSignal/Components/GradientTransformComponent.h [hunk 2]
```diff
@@ -130,6 +131,9 @@ namespace GradientSignal
         AZ::Vector3 GetBounds() const override;
         void SetBounds(AZ::Vector3 bounds) override;
 
+        AZ::Vector3 GetCenter() const override;
+        void SetCenter(AZ::Vector3 center) override;
```
Comment (moraaar) [inline Gems/GradientSignal/Code/Include/GradientSignal/Components/GradientTransformComponent.h:135]: The parameter in `SetCenter` is passed by value and therefore making copies. Any particular reason for not using const ref?

Same for `SetBounds`, `SetTranslate`, `SetRotate` and `SetScale`. It'd be nice to use const ref in all of them.

**Review by moraaar — APPROVED** [review]

File: Gems/GradientSignal/Code/Include/GradientSignal/Components/GradientTransformComponent.h (hunk 2 above)
Comment (greerdv) [reply Gems/GradientSignal/Code/Include/GradientSignal/Components/GradientTransformComponent.h:135]: I wondered about that too, but I just copied what all the other functions here were doing.

Comment (greerdv) [reply Gems/GradientSignal/Code/Include/GradientSignal/Components/GradientTransformComponent.h:135]: ok, can do.

Comment (greerdv) [reply Gems/GradientSignal/Code/Include/GradientSignal/Components/GradientTransformComponent.h]: updated

Comment (moraaar) [reply Gems/GradientSignal/Code/Include/GradientSignal/Components/GradientTransformComponent.h]: Thanks!

**Pull request merged**